src/ml_board/frontend/newDashboard/src/webworkers/worker_utils.tsx

```diff
@@ -0,0 +1,54 @@
+import SocketClass from '../websocket/SocketClass';
+import { handleEvaluationResultData, evalResultCustomData, evalResultSocketData } from './event_handlers/evaluationResultDataHandler';
+import EVENT_TYPE from './socketEventConstants';
+
+let webSocket = null;
+
+type parsedSocketData = {
+    creation_ts: BigInteger,
+    event_type: string,
+    payload: evalResultSocketData //as you handle experiment_status, job_status, keep adding: evalResultSocketData | expStatusSocketData | jobStatusSocketData like this & so on...
+}
+
+const workerOnMessageCallback = (evalResultCustomData:evalResultCustomData, parsedSocketData:parsedSocketData)=>{
+
+    let eventType = parsedSocketData["event_type"].toLowerCase();
+    let dataToUpdateReduxInChart = null;
+    let dataToUpdateReduxInDashboard = null
+    
+    switch(eventType) {
```

Comment:
I think we should replace the switch by a Dictionary mapping the `event_type` to the respective processing method. What do you think?